jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SchemaSelectionPanel.kt

```diff
@@ -3,142 +3,92 @@
-import software.aws.toolkits.jetbrains.services.schemas.SchemaTemplateExtraContext
-import software.aws.toolkits.jetbrains.services.schemas.SchemaTemplateParameters
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamSchemaDownloadPostCreationAction
+import software.aws.toolkits.jetbrains.services.schemas.SchemaCodeLangs
+import software.aws.toolkits.jetbrains.ui.connection.AwsConnectionSettingsSelector
+import software.aws.toolkits.resources.message
 import javax.swing.JComponent
-import javax.swing.JLabel
-import javax.swing.JPanel
 
-// UI for selecting a Schema
-interface SchemaSelectionPanel {
-    val schemaSelectionPanel: JComponent
-
-    val schemaSelectionLabel: JLabel?
-
-    fun registryName(): String? = null
-
-    fun schemaName(): String? = null
-
-    fun reloadSchemas(awsConnection: ConnectionSettings? = null) {}
-
-    fun buildSchemaTemplateParameters(): SchemaTemplateParameters?
-
-    fun validateAll(): List<ValidationInfo>? = null
-
-    companion object {
-
-        @JvmStatic
-        fun create(
-            runtime: Runtime,
-            selectedTemplate: SamProjectTemplate,
-            generator: SamProjectGenerator
-        ): SchemaSelectionPanel =
-            runtime.runtimeGroup?.let { runtimeGroup ->
-                if (selectedTemplate.supportsDynamicSchemas())
-                    SamProjectWizard.getInstance(runtimeGroup).createSchemaSelectionPanel(generator)
-                else
-                    NoOpSchemaSelectionPanel()
-            } ?: NoOpSchemaSelectionPanel()
+/*
+ * A panel encapsulating  AWS credential selection during SAM new project creation wizard
+  */
+class SchemaSelectionPanel : WizardFragment {
+    private val schemaSelector = SchemaResourceSelector()
+    private val awsConnectionSelector = AwsConnectionSettingsSelector(
+        DefaultProjectFactory.getInstance().defaultProject,
+        schemaSelector::reloadSchemas
+    )
+    private val component = panel {
+        row {
+            awsConnectionSelector.selectorPanel()(grow)
+        }
+        row(message("sam.init.schema.label")) {
+            schemaSelector.component(grow)
+        }
     }
-}
-
-// UI-agnostic schema selection panel
-abstract class SchemaSelectionPanelBase(private val project: Project) :
-    SchemaSelectionPanel {
 
-    private val schemaDownloader = SchemaDownloader()
+    override fun title(): String = message("sam.init.schema.label")
 
-    override fun buildSchemaTemplateParameters(): SchemaTemplateParameters? {
-        val schemaName = schemaName()
-        val registryName = registryName()
+    override fun component(): JComponent = component
 
-        if (schemaName == null || registryName == null) {
-            return null
+    override fun validateFragment(): ValidationInfo? {
+        if (awsConnectionSelector.selectedCredentialProvider() == null) {
```

Comment:
elvis again?

Reply:
Since I do not need the result, not sure elvis makes sense